include/sdfg/debug_info.h

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <cassert>
+#include <cstddef>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+namespace sdfg {
+
+namespace builder {
+class SDFGBuilder;
+class StructuredSDFGBuilder;
+} // namespace builder
+
+namespace serializer {
+class JSONSerializer;
+} // namespace serializer
+
+class Function;
+
+struct DebugLoc {
+    std::string filename_;
```

Comment:
Public attributes don't have "_", see Google coding style. Or use public methods and private members